include/deal2lkit/parsed_preconditioner/amg_muelu.h

```diff
@@ -0,0 +1,173 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2015 by the deal2lkit authors
+//
+//    This file is part of the deal2lkit library.
+//
+//    The deal2lkit library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal2lkit distribution.
+//
+//-----------------------------------------------------------
+
+#ifndef _d2k_parsed_amg_muelu_preconditioner_h
+#define _d2k_parsed_amg_muelu_preconditioner_h
+
+#include <deal2lkit/config.h>
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_TRILINOS
+
+#include <deal.II/lac/trilinos_precondition.h>
+
+#include <deal2lkit/parsed_finite_element.h>
+#include <deal2lkit/parameter_acceptor.h>
+#include <deal2lkit/utilities.h>
+
+using namespace dealii;
+
+
+D2K_NAMESPACE_OPEN
+
+/**
+ * A parsed AMG preconditioner which uses parameter files to choose
```

Comment:
@asartori86 
`AMGMueLu` uses [MueLu](https://trilinos.org/packages/muelu/). I made some experiments and I have noticed that `AMG` and `AMGMueLo` perform different, even if you use same parameters ( `AMG` has just one parameter more than `AMGMueLu`,  that is `higher_order_elements` ).

I was curious to see if this preconditioner would perform better on a cluster:
> MueLu is designed to solve large sparse linear systems of equations arising from PDE discretizations. MueLu provides easy-to-use multigrid solvers and preconditioners based on smoothed aggregation algorithms. As a multigrid framework, MueLu supports the design of highly application specific multigrid preconditioners. MueLu has been used for pressure solves within a low Mach thermal fluids simulation on 524K cores of a BlueGene/Q machine, and for solving a fully-coupled MHD problem on 524K BlueGene/Q cores.  

(From the website of [MueLu](https://trilinos.org/packages/muelu/))